tests/cata_utility_test.cpp

```diff
@@ -306,3 +306,26 @@ TEST_CASE( "check_debug_menu_string_methods", "[debug_menu]" )
         CHECK( pair.first == joined );
     }
 }
+
+TEST_CASE( "lcmatch", "[utility]" )
+{
+    CHECK( lcmatch( "bo", "bo" ) == true );
+    CHECK( lcmatch( "Bo", "bo" ) == true );
+    CHECK( lcmatch( "Bo", "co" ) == false );
+
+    CHECK( lcmatch( "Bö", "bo" ) == true );
+    CHECK( lcmatch( "Bō", "bo" ) == true );
+    CHECK( lcmatch( "BÖ", "bo" ) == true );
+    CHECK( lcmatch( "BŌ", "bo" ) == true );
+    CHECK( lcmatch( "Bö", "co" ) == false );
+    CHECK( lcmatch( "Bō", "co" ) == false );
+    CHECK( lcmatch( "BÖ", "co" ) == false );
+    CHECK( lcmatch( "BŌ", "co" ) == false );
+
+    CHECK( lcmatch( "«101 борцовский приём»", "при" ) == true );
+    CHECK( lcmatch( "«101 борцовский приём»", "прИ" ) == true );
+    CHECK( lcmatch( "«101 борцовский приём»", "прб" ) == false );
+
+    CHECK( lcmatch( "無効", "無" ) == true );
+    CHECK( lcmatch( "無効", "無效" ) == false );
+}
```

Comment:
I suggest adding
`CHECK( lcmatch( "bO", "bo" ) == true );`
and
`CHECK( lcmatch( "bO", "bō" ) == true );`

Reply:
I thought about this, and I think only accents from the data string should be removed for the fuzzy search. If the user types in "bo" as search key, it should match with "Bo", "Bö" and "Bō"; but if the user keys in "bö", I believe the user specifically wants to search for "Bö" or "BÖ".

Reply:
I meant add a check for small first letter and capitalized second letter.

Reply:
My previous reply was in response to the second check you proposed where I now think it should be `CHECK( lcmatch( "bO", "bō" ) == false ); `

Because I tested case conversion by the pair of Bo and bo so I don’t it’s necessary to test it again because the position does not matter.